assoc-ty.chalk

```diff
@@ -0,0 +1,29 @@
+trait Iterator {
+    type Item;
+}
+
+struct Chars { }
+
+struct char { }
+
+impl Iterator for Chars {
+    type Item = char;
+}
+
+struct Vec<T> { }
+
+impl<T> Iterator for Vec<T> {
+    type Item = T;
+}
+
+// trait IntoIterator where <Self::IntoIter as Iterator>::Item = Self::Item {
+//     type Item;
+//     type IntoIter: Iterator;
+// }
+// 
+// struct IntoIter<T> { }
+// 
+// impl<T> IntoIterator for Vec<T> {
+//     type Item = T;
+//     type IntoIter = IntoIter<T>;
+// }
```

Comment:
we don't usually add random `.chalk` files into the repo (maybe I should make an `examples` directory or something though)

Reply:
Whoops, forgot to remove that from the commit...